rustls/src/verify.rs

```diff
@@ -321,6 +321,70 @@ impl fmt::Debug for dyn ClientCertVerifier {
     }
 }
 
+/// Verify that the end-entity certificate `end_entity` is a valid server cert
+/// and chains to at least one of the [OwnedTrustAnchor] in the `roots` [RootCertStore].
+///
+/// `intermediates` contains all certificates other than `end_entity` that
+/// were sent as part of the server's [Certificate] message. It is in the
+/// same order that the server sent them and may be empty.
+#[allow(unreachable_pub, dead_code)]
+#[cfg_attr(docsrs, doc(cfg(feature = "dangerous_configuration")))]
+pub fn verify_server_cert_signed_by_trust_anchor(
+    cert: &ParsedCertificate,
+    roots: &RootCertStore,
+    intermediates: &[Certificate],
+    now: SystemTime,
+) -> Result<(), Error> {
+    let chain: Vec<&[u8]> = intermediates
```

Comment:
Could we reuse `prepare()` here, maybe by changing its API a bit to fit better?

Reply:
I split it to seperate functions for each argument